pandas/formats/format.py

```diff
@@ -1756,13 +1758,501 @@ def __init__(self, row, col, val, style=None, mergestart=None,
+            try:
+                val = float(val)
+            except ValueError:
+                warnings.warn('Unhandled font size: %r' % val + unit,
+                              CSSWarning)
+
+        while unit != 'pt':
+            if unit == 'em':
+                if em_pt is None:
+                    unit = 'rem'
+                else:
+                    val *= em_pt
+                    unit = 'pt'
+                continue
+
+            try:
+                unit, mul = conversions[unit]
+            except KeyError:
+                warnings.warn('Unknown size unit: %r' % unit, CSSWarning)
+                return self.size_to_pt('1!!default', conversions=conversions)
+            val *= mul
+
+        val = round(val, 5)
+        if int(val) == val:
+            size_fmt = '%d'
+        else:
+            size_fmt = '%f'
+        return (size_fmt + 'pt') % val
+
+    def atomize(self, declarations):
+        for prop, value in declarations:
+            attr = 'expand_' + prop.replace('-', '_')
+            try:
+                expand = getattr(self, attr)
+            except AttributeError:
+                yield prop, value
+            else:
+                for prop, value in expand(prop, value):
+                    yield prop, value
+
+    SIDE_SHORTHANDS = {
+        1: [0, 0, 0, 0],
+        2: [0, 1, 0, 1],
+        3: [0, 1, 2, 1],
+        4: [0, 1, 2, 3],
+    }
+    SIDES = ('top', 'right', 'bottom', 'left')
+
+    def _side_expander(prop_fmt):
+        def expand(self, prop, value):
+            tokens = value.split()
+            try:
+                mapping = self.SIDE_SHORTHANDS[len(tokens)]
+            except KeyError:
+                warnings.warn('Could not expand "%s: %s"' % (prop, value),
+                              CSSWarning)
+                return
+            for key, idx in zip(self.SIDES, mapping):
+                yield prop_fmt % key, tokens[idx]
+
+        return expand
+
+    expand_border_color = _side_expander('border-%s-color')
+    expand_border_style = _side_expander('border-%s-style')
+    expand_border_width = _side_expander('border-%s-width')
+    expand_margin = _side_expander('margin-%s')
+    expand_padding = _side_expander('padding-%s')
+
+    def parse(self, declarations_str):
+        """Generates (prop, value) pairs from declarations
+
+        In a future version may generate parsed tokens from tinycss/tinycss2
+        """
+        for decl in declarations_str.split(';'):
+            if not decl.strip():
+                continue
+            prop, sep, val = decl.partition(':')
+            prop = prop.strip().lower()
+            # TODO: don't lowercase case sensitive parts of values (strings)
+            val = val.strip().lower()
+            if not sep:
+                warnings.warn('Ill-formatted attribute: expected a colon '
+                              'in %r' % decl, CSSWarning)
+            yield prop, val
+
+
+class CSSToExcelConverter(object):
```

Comment:
this class maybe put in ``pandas/formats/excel.py`` (just trying to have a module be clear / do 1 thing.